src/core/handlers/GraphQLHandler.ts

```diff
@@ -141,11 +144,17 @@ export class GraphQLHandler extends RequestHandler<
     const match = matchRequestUrl(new URL(args.request.url), this.endpoint)
     if (!match.matches) return { match }
 
-    const parsedResult = await parseGraphQLRequest(args.request).catch(
-      (error) => {
-        console.error(error)
-        return undefined
-      },
+    if (!GraphQLHandler.#parseGraphQLRequestCache.has(args.request)) {
+      GraphQLHandler.#parseGraphQLRequestCache.set(
```

Comment:
not the best place to do this, since we don't pass to onUnhandledRequest well, but gets rid of the `O(n)` cloning at least

Reply:
I don't think `onUnhandledRequest` reads the request body to require a clone, does it? Maybe we shouldn't concern ourselves about it. If the consumer wants to read the request/response body, they can clone it in the `onUnhandledRequest` callback. Unless there's a timing issue, of course, and that clone should've been made prior to that callback being called.

Reply:
> I don't think onUnhandledRequest reads the request body to require a clone, does it

yep!  it does this to determine which handlers might have matched - https://github.com/mswjs/msw/blob/ea28bd9ad2efe0918660cd37054f21883caa3d30/src/core/utils/request/onUnhandledRequest.ts#L144-L146

it's not a big deal, as this is only 1 extra clone per request cycle, but we will have already done this parsing once, potentially, so it's also possibly a wasted clone, where a cached `parseGraphQLRequest` will avoid that